Frameworks/CoreLocation/CLPlacemark.mm

```diff
@@ -14,25 +15,117 @@
 //
 //******************************************************************************
 
-#import <StubReturn.h>
 #import <CoreLocation/CLPlacemark.h>
+#import <Starboard.h>
+#import <StubReturn.h>
+#import <AddressBook/ABPerson.h>
+
+@interface CLPlacemark()
+
+@property (readwrite, nonatomic) CLLocation* location;
+@property (readwrite, nonatomic) NSString* name;
+@property (readwrite, nonatomic) NSDictionary* addressDictionary;
+@property (readwrite, nonatomic) NSString* ISOcountryCode;
+@property (readwrite, nonatomic) NSString* country;
+@property (readwrite, nonatomic) NSString* postalCode;
+@property (readwrite, nonatomic) NSString* administrativeArea;
+@property (readwrite, nonatomic) NSString* subAdministrativeArea;
+@property (readwrite, nonatomic) NSString* locality;
+@property (readwrite, nonatomic) NSString* subLocality;
+@property (readwrite, nonatomic) NSString* thoroughfare;
+@property (readwrite, nonatomic) NSString* subThoroughfare;
+@property (readwrite, nonatomic) CLRegion* region;
+@property (readwrite, nonatomic) NSTimeZone* timeZone;
+@property (readwrite, nonatomic) NSString* inlandWater;
+@property (readwrite, nonatomic) NSString* ocean;
+@property (readwrite, nonatomic) NSArray<NSString*>* areasOfInterest;
+
+@end
 
 @implementation CLPlacemark
 
+- (instancetype)initWithName:(NSString*)name location:(CLLocation*)location {
+    if (self = [super init]) {
+        _name = name;
+        _location = location;
+    }
+
+    return self;
+}
+
+- (instancetype)initWithLocation:(CLLocation*)location
+                dictionary:(NSMutableDictionary*)dictionary
+{
+    if (self = [super init]) {
+        _location = location;
+        _name = [dictionary objectForKey:@"placemarkName"];
+        _ISOcountryCode = [dictionary objectForKey:@"placemarkISOcountryCode"];
+        _country = [dictionary objectForKey:@"placemarkCountry"];
+        _postalCode = [dictionary objectForKey:@"placemarkPostalCode"];
+        _administrativeArea = [dictionary objectForKey:@"placemarkAdministrativeArea"];
+        _subAdministrativeArea = [dictionary objectForKey:@"placemarkSubAdministrativeArea"];
+        _locality = [dictionary objectForKey:@"placemarkLocality"];
+        _subLocality = [dictionary objectForKey:@"placemarkSubLocality"];
+        _thoroughfare = [dictionary objectForKey:@"placemarkThoroughfare"];
+        _subThoroughfare = [dictionary objectForKey:@"placemarkSubThoroughfare"];
```

Comment:
same here. It seems that we need to somehow say that we are taking ownership here.